File: app/src/test/java/dk/itu/trashapp/TestItemsDB.java
package dk.itu.trashapp;

import org.junit.Test;
import org.junit.Before;
import static org.junit.Assert.*;

public class TestItemsDB {
    private ItemsDB myItemsDB;

    @Before
    public void setUp(){
        myItemsDB = ItemsDB.get();
    }

    @Test
    public void get_MultipleCalls_ReturnsSameInstance() {
        ItemsDB instance1 = ItemsDB.get();
        ItemsDB instance2 = ItemsDB.get();
        assertSame(instance1, instance2);
      }

    @Test
    public void addItem_WhenAddingItem_ItemDBIsUpdated(){
        String what = "bottle";
        String where = "recycling";
        myItemsDB.addItem(what, where);
        assertEquals(where, myItemsDB.search(what));
    }

    @Test
    public void search_WhenParamNotInDB_ReturnsNotFound(){
        String searchResult = myItemsDB.search("");
        assertEquals("not found", searchResult);
    }

    @Test
    public void search_WhenKeyExists_ReturnsNotFound(){
        myItemsDB.addItem("carrot", "food");
        String searchResult = myItemsDB.search("carrot");
        assertEquals("food", searchResult);
    }

}

File: app/src/test/java/dk/itu/trashapp/TestItemsDB.kt
package dk.itu.trashapp

import dk.itu.trashapp.ItemsDB.Companion.get
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import java.io.IOException

class TestItemsDB {
  /*  private var myItemsDB: ItemsDB? = null

    @Before
    @Throws(IOException::class)
    fun setUp() {
        myItemsDB = get()
    }

    @Test
    @Throws(IOException::class)
    fun get_MultipleCalls_ReturnsSameInstance() {
        val instance1 = get()
        val instance2 = get()
        Assert.assertSame(instance1, instance2)
    }

    @Test
    @Throws(IOException::class)
    fun addItem_WhenAddingItem_ItemDBIsUpdated() {
        val what = "bottle"
        val where = "recycling"
        myItemsDB!!.addItem(what, where)
        Assert.assertEquals(where, myItemsDB!!.search(what))
    }

    @Test
    fun search_WhenParamNotInDB_ReturnsNotFound() {
        val searchResult = myItemsDB!!.search("")
        Assert.assertEquals("not found", searchResult)
    }

    @Test
    @Throws(IOException::class)
    fun search_WhenKeyExists_ReturnsNotFound() {
        myItemsDB!!.addItem("carrot", "food")
        val searchResult = myItemsDB!!.search("carrot")
        Assert.assertEquals("food", searchResult)
    }

    */
}
